internal/plugin/connectors/tcp/pg/protocol/protocol_test.go

```diff
@@ -0,0 +1,63 @@
+package protocol
+
+import (
+	"encoding/binary"
+	"net"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func TestReadMessage(t *testing.T) {
+	t.Run("parses contents", func(t *testing.T) {
+		r, w := net.Pipe()
+		expectedMessageType := byte(12)
+		expectedMessage := []byte{0,1,2,3,4}
+
+		go func() {
+			err := binary.Write(w, binary.BigEndian, expectedMessageType)
+			if err != nil {
+				panic(err)
+			}
+			err = binary.Write(w, binary.BigEndian, int32(len(expectedMessage) + 4))
+			if err != nil {
+				panic(err)
+			}
+
+			_, err = w.Write(expectedMessage)
+			if err != nil {
+				panic(err)
+			}
+		}()
+		messageType, message, err := ReadMessage(r)
+
+		if !assert.NoError(t, err) {
+			return
+		}
+
+		assert.Equal(t, expectedMessage, message)
+		assert.Equal(t, expectedMessageType, messageType)
+	})
+
+	t.Run("validates message length", func(t *testing.T) {
```

Comment:
I may be mistaken, but it looks like we're missing a test case. It could fail to write, or it could fail because the length is wrong. Shouldn't we have two separate fail case tests?
>  Add unit tests for readMessage that at least verify its behavior when the message length is valid, invalid but > 4, or invalid and < 4